Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/SpinBox.h

```diff
@@ -35,10 +38,87 @@ namespace AzQtComponents
         class SpinBoxLineEdit;
     }
 
+    namespace internal
+    {
+        /*
+            A small helper for deferring `setValue' events while a user is editing a textbox-like property.
+            In these cases, any external `setValue' events shouldn't overwrite what the user enters.
+        */
+        template<class T>
+        class DeferredTextboxLikeEdit
+        {
+        protected:
+            // Call to check if you should really set the new value
+            // This might record that value for later use
+            bool shouldReallySetValue(const T& prevValue, const T& newValue, bool isUserEditing);
```

Comment:
nit: This could use `//!` style too as they're more API comments than in-line guide comments

Would it be possible to also add in the comment what the return value indicates (I can probably guess but would be great to have it be explicit).

Reply:
(repeated in a few spots in this file) - love the comments themselves by the way 🙂

Reply:
added a bit more detail to comments here and used `//!`